providers/dns/autodns/autodns.go

```diff
@@ -0,0 +1,116 @@
+package autodns
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/go-acme/lego/v3/challenge/dns01"
+	"github.com/go-acme/lego/v3/platform/config/env"
+)
+
+const (
+	envAPIUser            = `AUTODNS_API_USER`
+	envAPIPassword        = `AUTODNS_API_PASSWORD`
+	envAPIEndpoint        = `AUTODNS_ENDPOINT`
+	envAPIEndpointContext = `AUTODNS_CONTEXT`
+	envTTL                = `AUTODNS_TTL`
+
+	defaultEndpoint = `https://api.autodns.com/v1/`
+	demoEndpoint    = `https://api.demo.autodns.com/v1/`
+
+	defaultEndpointContext int = 4
+	defaultTTL             int = 600
+)
+
+type Config struct {
+	Endpoint   *url.URL
+	Username   string `json:"username"`
+	Password   string `json:"password"`
+	Context    int    `json:"-"`
+	TTL        int    `json:"-"`
+	HTTPClient *http.Client
+}
+
+func NewDefaultConfig() *Config {
+	endpoint, _ := url.Parse(defaultEndpoint)
+
+	return &Config{
+		Endpoint:   endpoint,
+		Context:    defaultEndpointContext,
+		HTTPClient: &http.Client{},
```

Comment:
could you follow the same pattern as other provider:
https://github.com/go-acme/lego/blob/a5a29187fea9f555ea43fae16ca65066c452d330/providers/dns/cloudflare/cloudflare.go#L32-L41

Reply:
Done.